### PR TITLE
Tests: TimeSeries Subscriptions

### PR DESCRIPTION
**Pull request by doctrino**
## Issue - TimeSeries can only be used in 10 subscriptions
image

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: tests/tests_integration/test_api/test_datapoint_subscriptions.py [hunk 1]
```diff
@@ -20,7 +21,9 @@
 )
 from cognite.client.utils._text import random_string
 
-TIMESERIES_EXTERNAL_IDS = [f"PYSDK DataPoint Subscription Test {no}" for no in range(20)]
+TIMESERIES_EXTERNAL_IDS = [
+    f"PYSDK DataPoint Subscription {platform.system()} {platform.python_version()} Test {no}" for no in range(20)
+]
```
Comment (haakonvt) [inline tests/tests_integration/test_api/test_datapoint_subscriptions.py:26]: May I suggest we make this a fixture?
```suggestion
def time_series_external_ids(os_and_py_version: str) -> list[str]:
    return [
        f"PYSDK DataPoint Subscription {os_and_py_version} Test {no}" for no in range(20)
    ]
```

File: tests/tests_integration/test_api/test_datapoint_subscriptions.py [hunk 2]
```diff
@@ -261,10 +264,14 @@ def test_iterate_data_subscription_datapoints_added(
             assert batch.subscription_changes.added[0].external_id == time_series_external_ids[0]
 
             existing_data = cognite_client.time_series.data.retrieve_dataframe(external_id=time_series_external_ids[0])
+            if existing_data.empty:
+                start = pd.Timestamp("now")
+            else:
+                start = existing_data.index[-1] + pd.Timedelta("1d")
```
Comment (haakonvt) [inline tests/tests_integration/test_api/test_datapoint_subscriptions.py:270]: This seems like bootstrapping the time series with at least one datapoint? I think that belongs in a fixture!

**Review by haakonvt — COMMENTED** [review]
I really hope this addresses the quite flaky dps sub tests 😄

File: tests/tests_integration/test_api/test_datapoint_subscriptions.py (hunk 1 above)
Comment (doctrino) [reply tests/tests_integration/test_api/test_datapoint_subscriptions.py:26]: Sure can do that. What is the main motivation as it is just constants anyways?

Comment (haakonvt) [reply tests/tests_integration/test_api/test_datapoint_subscriptions.py:26]: I agree it is not necessary, but you get `os_an_py_version` for free which is nice 🤓

File: tests/tests_integration/test_api/test_datapoint_subscriptions.py (hunk 2 above)
Comment (doctrino) [reply tests/tests_integration/test_api/test_datapoint_subscriptions.py:267]: This I am more skeptical about. If it is only used in one test, moving it to a fixture seems to obfuscate.

**Review by haakonvt — APPROVED** [review]

File: tests/tests_integration/test_api/test_datapoint_subscriptions.py (hunk 2 above)
Comment (haakonvt) [reply tests/tests_integration/test_api/test_datapoint_subscriptions.py]: The test is using a random xid (see fixture `time_series_external_ids`), meaning all these xids should have an existing datapoint. Anyway, approving 👍

**Pull request merged**